src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/KubernetesCredentialsStore.java

```diff
@@ -23,16 +24,18 @@ public class KubernetesCredentialsStore extends CredentialsStore {
 
     private final KubernetesCredentialProvider provider;
     private final KubernetesCredentialsStoreAction action = new KubernetesCredentialsStoreAction(this);
+    private final ItemGroup<?> context;
 
-    public KubernetesCredentialsStore(KubernetesCredentialProvider provider) {
+    public KubernetesCredentialsStore(KubernetesCredentialProvider provider, ItemGroup<?> context) {
```

Comment:
if we can restrict credentials to Itesm (which it appears we can) then should this not be `Item` not ItemGroup?
(where `null` will represent Jenkins - which is normal)?

Reply:
i agree with going for item instead of itemgroup. but why would null represent jenkins?

Reply:
> i agree with going for item instead of itemgroup. but why would null represent jenkins?

Jenkins is not an Item (but is an ItemGroup) so can not be represented here.

but also credentials is generally treating null as `Jenkins`    (we are using an older version of the API here currently without those method overaloads but) 

https://github.com/jenkinsci/credentials-plugin/blob/cf0a900b37c29359c89e0d85fdc48361ef4c0eb1/src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java#L371C61-L371C61

https://github.com/jenkinsci/credentials-plugin/blob/cf0a900b37c29359c89e0d85fdc48361ef4c0eb1/src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java#L540-L542

so the absence of context means you are in the context of Jenkins.

now we could say - there is currently no other provider that I know of that can scope credentials to items (only ItemGroups) so we could say this is moot and using an Item is over-engineering, and punt all of that to a later enhancement.

Reply:
I think this then would require a lot of changes in the provider and the methods we are using. I would stay with ItemGroup as it seems out of scope for this.